From: Mike Hommey <mh@glandium.org>
Date: Mon, 16 Dec 2013 10:53:58 +0900
Subject: Add debian/extra-stuff to upstream build system directory traversal

---
 moz.build | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/moz.build b/moz.build
index 69ab963..c28bab2 100644
--- a/moz.build
+++ b/moz.build
@@ -203,4 +203,7 @@ SPHINX_TREES["gtest"] = "docs/gtest"
 
 SPHINX_TREES["nspr"] = "docs/nspr"
 
+if not CONFIG["LIBXUL_SDK"]:
+    DIRS += ["debian/extra-stuff"]
+
 include("build/templates.mozbuild")
