From: Peter Michael Green <plugwash@raspbian.org>
Date: Tue, 15 Feb 2022 22:36:23 +0000
X-Dgit-Generated: 91.6.0esr-1~deb10u1+rpi1 3e5d3e6733517cc1fdd81fc9eb445c8f52410d51
Subject: Cast parameters to makestack to uintptr_t which is not always the same as unsigned long.


---

--- firefox-esr-91.6.0esr.orig/toolkit/components/telemetry/tests/gtest/TestCombinedStacks.cpp
+++ firefox-esr-91.6.0esr/toolkit/components/telemetry/tests/gtest/TestCombinedStacks.cpp
@@ -87,10 +87,10 @@ TEST(CombinedStacks, Combine)
   //     02 modZ+2000
   //     03 base+4000
   const ProcessedStack testStacks[] = {
-      MakeStack(moduleSet1, {100ul, 200ul, 300ul}),
-      MakeStack(moduleSet1, {1000ul, 2000ul, 3000ul}),
-      MakeStack(moduleSet2, {100ul, 200ul, 300ul, 400ul}),
-      MakeStack(moduleSet2, {1000ul, 2000ul, 3000ul, 4000ul}),
+      MakeStack(moduleSet1, {(uintptr_t)100ul, (uintptr_t)200ul, (uintptr_t)300ul}),
+      MakeStack(moduleSet1, {(uintptr_t)1000ul, (uintptr_t)2000ul, (uintptr_t)3000ul}),
+      MakeStack(moduleSet2, {(uintptr_t)100ul, (uintptr_t)200ul, (uintptr_t)300ul, (uintptr_t)400ul}),
+      MakeStack(moduleSet2, {(uintptr_t)1000ul, (uintptr_t)2000ul, (uintptr_t)3000ul, (uintptr_t)4000ul}),
   };
 
   // combined1 <-- testStacks[0] + testStacks[1]
