From: Peter Michael Green <plugwash@raspbian.org>
Date: Wed, 7 Oct 2020 19:41:49 +0000
X-Dgit-Generated: 78.3.0esr-2+rpi1 b1f30fd7ea0b729a7bbc7dc1e804e2a0cc0c573a
Subject: Disable neon in qcms.


---

diff --git a/gfx/qcms/qcmsint.h b/gfx/qcms/qcmsint.h
index b08fc9490b..3f5ad40145 100644
--- a/gfx/qcms/qcmsint.h
+++ b/gfx/qcms/qcmsint.h
@@ -96,18 +96,18 @@ void qcms_transform_data_bgra_out_lut_altivec(const qcms_transform *transform,
                                               unsigned char *dest,
                                               size_t length);
 
-void qcms_transform_data_rgb_out_lut_neon(const qcms_transform *transform,
-                                          const unsigned char *src,
-                                          unsigned char *dest,
-                                          size_t length);
-void qcms_transform_data_rgba_out_lut_neon(const qcms_transform *transform,
-                                           const unsigned char *src,
-                                           unsigned char *dest,
-                                           size_t length);
-void qcms_transform_data_bgra_out_lut_neon(const qcms_transform *transform,
-                                           const unsigned char *src,
-                                           unsigned char *dest,
-                                           size_t length);
+//void qcms_transform_data_rgb_out_lut_neon(const qcms_transform *transform,
+//                                          const unsigned char *src,
+//                                          unsigned char *dest,
+//                                          size_t length);
+//void qcms_transform_data_rgba_out_lut_neon(const qcms_transform *transform,
+//                                           const unsigned char *src,
+//                                           unsigned char *dest,
+//                                           size_t length);
+//void qcms_transform_data_bgra_out_lut_neon(const qcms_transform *transform,
+//                                           const unsigned char *src,
+//                                           unsigned char *dest,
+//                                           size_t length);
 
 extern bool qcms_supports_iccv4;
 extern bool qcms_supports_neon;
