From: Peter Michael Green <plugwash@raspbian.org>
Date: Tue, 21 Dec 2021 17:35:22 +0000
Subject: Use a #define instead of a typedef for double_t in fdlibm to prevent conflicting definitions error

diff --git a/modules/fdlibm/src/math_private.h b/modules/fdlibm/src/math_private.h
index 632d2c4c619..b7fecb02e58 100644
--- a/modules/fdlibm/src/math_private.h
+++ b/modules/fdlibm/src/math_private.h
@@ -35,7 +35,7 @@ typedef long double      __double_t;
 #else
 typedef double      __double_t;
 #endif
-typedef __double_t  double_t;
+#define double_t __double_t
 typedef float       __float_t;
 
 /*
