From: Mike Hommey <mh@glandium.org>
Date: Fri, 10 Jun 2022 06:22:33 +0900
Subject: Allow to build with cargo in Debian unstable

---
 build/moz.configure/rust.configure | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/build/moz.configure/rust.configure b/build/moz.configure/rust.configure
index e64dc5d..10c8947b 100644
--- a/build/moz.configure/rust.configure
+++ b/build/moz.configure/rust.configure
@@ -171,7 +171,7 @@ def rust_compiler(rustc_info, cargo_info, target):
             )
         )
     rustc_min_version = Version(MINIMUM_RUST_VERSION)
-    cargo_min_version = rustc_min_version
+    cargo_min_version = Version("1.56.0")
 
     version = rustc_info.version
     is_nightly = "nightly" in version.version
