--- firefox-esr-68.9.0esr.orig/build/moz.configure/rust.configure
+++ firefox-esr-68.9.0esr/build/moz.configure/rust.configure
@@ -346,6 +346,8 @@
             return None
 
         rustc_target = find_candidate(candidates)
+        if rustc_target.startswith('i586'):
+            rustc_target = rustc_target.replace('i586', 'i686')
 
         if rustc_target is None:
             die("Don't know how to translate {} for rustc".format(
