From: Mike Hommey <mh@glandium.org>
Date: Sun, 2 Apr 2017 06:33:19 +0900
Subject: Use the Mozilla Location Service key when the Google Key is not there

---
 modules/libpref/init/all.js | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/modules/libpref/init/all.js b/modules/libpref/init/all.js
index f0f3449b..86fe1304 100644
--- a/modules/libpref/init/all.js
+++ b/modules/libpref/init/all.js
@@ -3072,7 +3072,12 @@ pref("network.tcp.keepalive.idle_time", 600); // seconds; 10 mins
 // All the Geolocation preferences are here.
 //
 #if !defined(ANDROID) && !defined(XP_WIN)
+#if defined(GOOGLE_LOCATION_SERVICE_API_KEY)
   pref("geo.provider.network.url", "https://www.googleapis.com/geolocation/v1/geolocate?key=%GOOGLE_LOCATION_SERVICE_API_KEY%&solution_channel=%OS%");
+#else
+  // Use MLS on Nightly and early Beta.
+  pref("geo.provider.network.url", "https://location.services.mozilla.com/v1/geolocate?key=%MOZILLA_API_KEY%");
+#endif
 
   // Timeout to wait before sending the location request.
   pref("geo.provider.network.timeToWaitBeforeSending", 5000);
